tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -1547,6 +1547,51 @@ static LogicalResult Verify(DynamicSliceOp op) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// RealDynamicSliceOp
+//===----------------------------------------------------------------------===//
+// Verifies that operand rank matches start_indices/limit_indices/strides size
+static LogicalResult Verify(RealDynamicSliceOp op) {
+  auto input_type = op.operand().getType().dyn_cast<RankedTensorType>();
+  if (!input_type) {
+    return op.emitOpError() << "operand is not RankedTensorType";
+  }
```

Comment:
Is the rejection of IR with unranked tensors something specific to this op or something that you think should be generalized to MHLO? If this is specific then please document it in the code, if this is to be done everywhere then we should start a thread on the mailing-list to discuss this, I'm not sure about the desired representational aspect here.

This also would be better achieved in ODS by introducing a new definition than HLO_Tensor (like `HLO_RankedTensor` for example)

Reply:
You are right. Op def in mhlo should be general. So now we make it return success() if the input is unranked tensor and let the lower dialect to handle it.